### PR TITLE
Fixed #33172 -- Added example of modifying upload handlers on the fly for CBVs.

### PR DESCRIPTION
**Pull request by sreeharicodes**
## Document CBV decoration for modifying upload handlers on the fly.

Ticket [#33172](https://code.djangoproject.com/ticket/33172)
Screenshot 2021-10-12 at 1 31 06 PM

### CODE REVIEW

**Comment by sreehari1997** [comment]
@felixxm why the docs checks are failing?

**Comment by kezabelle** [comment]
So, first off, thanks for the patch :)
Secondly, let me apologise up front if this is a dumb question -- as I mentioned on the ticket, I genuinely couldn't tell you which bits need decorating. 

Is this truly _the right way_ to do it, or is it providing an exact-facsimile of the FBV for consistency? I ask because the FBV docs say _you should always modify uploading handlers as early in your view as possible_ ... for FBVs that's the first line of the callable, but for CBVs it's `setup(request, ...)` rather than `dispatch(request, ...)` (see [ticket 29750](https://code.djangoproject.com/ticket/29750), notionally for just this kind of thing, ignoring CSRF twiddling). Now, I can guess that perhaps `dispatch` _is_ the earliest point you can `csrf_exempt` because of the way `as_view()` and the `CSRFMiddleware` and `csrf_exempt` itself all work together, but I'm not **sure** so I'm asking.

Basically, if the goal is to demonstrate _how_ we want people to do it, is this the best case scenario, or is there something more ergonomic/pythonic/whatever available if we ditch parity with the FBV example.

**Comment by bobort** [comment]
This is great! Thank you so much for doing this! I've learned something new, and I was able to implement this according to this practice just fine. I cannot thank you enough!

**Comment by sreehari1997** [comment]
@felixxm @kezabelle yes.. it's better with `setup():`.
I will make the necessary changes.
